src/NuGet.Core/NuGet.Commands/RestoreCommand/Logging/WarningPropertiesCollection.cs

```diff
@@ -4,20 +4,28 @@
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Linq;
 using NuGet.Common;
 using NuGet.Frameworks;
 using NuGet.ProjectModel;
+using NuGet.Shared;
 
 namespace NuGet.Commands
 {
     /// <summary>
-    /// Class to hold ProjectWIde and PackageSpecific WarningProperties
+    /// Class to hold ProjectWide and PackageSpecific WarningProperties.
     /// </summary>
     public class WarningPropertiesCollection
     {
         private readonly ConcurrentDictionary<string, NuGetFramework> _getFrameworkCache = new ConcurrentDictionary<string, NuGetFramework>();
 
         /// <summary>
+        /// Contains the target frameworks for the project.
+        /// These are used for no warn filtering in case of a log message without a target graph.
+        /// </summary>
+        public IReadOnlyList<NuGetFramework> ProjectFrameworks { get; set; } = new List<NuGetFramework>();
```

Comment:
We dont have any convention for that. We use this style in other places across our code.